test/dynamo/test_misc.py

```diff
@@ -3027,6 +3096,7 @@ def f(pred, x):
         a = opt_fn(torch.tensor(False), torch.tensor([0.25, 0.25]))
         self.assertTrue(same(torch.tensor([1.25, 1.25]), a))
 
+    @unittest.expectedFailure
```

Comment:
how come?

Reply:
This failed by ```TypeError: forward() missing 1 required positional argument: 'self_w'```.  I chatted with @zhxchen17 and he mentioned side effect was not planned to be supported then, so this was an expected failure. I'm not sure how we should handle side effect under the current story. Should we check side effects from true/false or parent/body branches and throw exception if they are not identical?

Reply:
Not sure if the following comment is still valid:
https://github.com/pytorch/pytorch/blob/f5a0b31a956e7d77fa7eee13d206dfe84e125a48/torch/_dynamo/variables/torch.py#L996

Reply:
If this is intentional, you shouldn't xfail the test; instead, the old assertRaises should just be adjusted so that it catches the right exception